test/Runtime/type_discovery.swift

```diff
@@ -0,0 +1,46 @@
+// RUN: %target-run-simple-swift
+// REQUIRES: executable_test
+
+import StdlibUnittest
+
+protocol P { }
+struct A: P { }
+class B: P { }
+enum C: P { }
+struct D /* does not conform */ { }
+
+guard #available(SwiftStdlib 9999, *) else {
+  print("Skipping tests in \(#fileID) due to unavailability.")
+  return
+}
+
+// Don't use #dsohandle on Windows because __ImageBase is not linked properly
+// when building the test target and we get a link-time error trying to use it.
```

Comment:
Do you have the error message?  `__ImageBase` is a synthetic symbol that is provided by the linker.

Reply:
Yes, but I haven't had the opportunity to write up a bug report for it. It's trivial to reproduce: add a test like this one and use `#dsohandle`, then watch the sparks fly.